astropy/wcs/wcs.py

```diff
@@ -1031,6 +1032,16 @@ def _read_sip_kw(self, header):
 
             del header[str('A_ORDER')]
             del header[str('B_ORDER')]
+            ctype=[header['CTYPE{0}'.format(nax)] for nax in range(1, self.naxis + 1)]
+            if any([ctyp[-4 :] != '-SIP' for ctyp in ctype]):
+                message = """
+                         Inconsistent SIP distortion information in header.
+                         Detected SIP coefficients but CTYPE is missing "-SIP" suffix.
+                         Please consider correcting the header.
+                         SIP distortion component was included in the WCS. Use ``wcs_pix2world``
+                         instead of ``all_pix2world`` to ignore all distortion components.
```

Comment:
`wcs_pix2world` turns off not only SIP but also all other (D2IM, CPDIS) distortions which may or may not be what user wants: see https://github.com/astropy/astropy/issues/4803#issuecomment-214573200. Maybe it would be better to tell users to turn off SIP (only) by setting `WCS.sip = None`